edgelet/edgelet-docker/src/error.rs

```diff
@@ -127,16 +149,21 @@ impl From<DockerError<serde_json::Value>> for Error {
             DockerError::Serde(error) => Error {
                 inner: Error::from(error).context(ErrorKind::Docker),
             },
-            DockerError::ApiError(ref error) if error.code == StatusCode::NOT_FOUND => {
-                Error::from(ErrorKind::NotFound)
-            }
-            DockerError::ApiError(ref error) if error.code == StatusCode::CONFLICT => {
-                Error::from(ErrorKind::Conflict)
-            }
-            DockerError::ApiError(ref error) if error.code == StatusCode::NOT_MODIFIED => {
-                Error::from(ErrorKind::NotModified)
-            }
-            _ => Error::from(ErrorKind::DockerRuntime(err)),
+            DockerError::ApiError(error) => match error.code {
+                StatusCode::NOT_FOUND => match get_message(error) {
+                    Ok(message) => Error::from(ErrorKind::NotFound(message)),
+                    Err(error) => {
+                        Error::from(ErrorKind::DockerRuntime(DockerError::ApiError(error)))
+                    }
```

Comment:
Can we do the same `map`/`unwrap_or_else` here (like the clause below?

Reply:
thanks.